### PR TITLE
Support iks nodes

### PR DESCRIPTION
**Pull request by adisos**

### CODE REVIEW

**Comment by kyorav** [comment]
It would help to document whatever assumptions and limitations we have for this version in the linked issue. Like does this first version support both all_subnets and all_endpoints? What are the assumptions about SGs? Etc.

**Review by zivnevo — CHANGES_REQUESTED** [review]
👍 A few small comments

File: pkg/ibmvpc/examples/iks_config_object_with_grouping.txt
[the remark this answers is not in the document]
Comment (adisos) [reply pkg/ibmvpc/examples/iks_config_object_with_grouping.txt:1]: actually the grouping for nodes is currently only per nodes of the same subnet, and on this example every iks-nodes is on a different subnet..

File: pkg/ibmvpc/parser.go [hunk 1]
```diff
@@ -113,6 +120,85 @@ func addParsedInstances(vList []json.RawMessage, res *ResourcesContainer) error
...
 }
 
+type iksNode struct {
+	Cidr      string
+	IPAddress string
+	SubnetID  string
+	ID        string
+}
+
+const iksParsingIssue = "issue parsing iks node"
```
[the remark this answers is not in the document]
Comment (adisos) [reply pkg/ibmvpc/parser.go]: done

File: pkg/ibmvpc/parser.go [hunk 2]
```diff
@@ -113,6 +120,85 @@ func addParsedInstances(vList []json.RawMessage, res *ResourcesContainer) error
...
+*/
+
+func parseIKSNode(m map[string]json.RawMessage) (*iksNode, error) {
+	res := &iksNode{}
+
+	// parse the "networkInterfaces" section
+	if _, ok := m[networkInterfaces]; !ok {
+		return nil, errors.New(iksParsingIssue)
+	}
+	netInterfaces, err := JSONToList(m[networkInterfaces])
```
[the remark this answers is not in the document]
Comment (adisos) [reply pkg/ibmvpc/parser.go]: done

File: pkg/ibmvpc/parser.go [hunk 3]
```diff
@@ -113,6 +120,85 @@ func addParsedInstances(vList []json.RawMessage, res *ResourcesContainer) error
...
+		return nil, errors.New(iksParsingIssue)
+	}
+	netInterfaceStr, err := json.Marshal(netInterfaces[0])
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(netInterfaceStr, res)
+	if err != nil {
+		return nil, err
+	}
```
[the remark this answers is not in the document]
Comment (adisos) [reply pkg/ibmvpc/parser.go]: right, simplified

File: pkg/ibmvpc/parser.go [hunk 4]
```diff
@@ -113,6 +120,85 @@ func addParsedInstances(vList []json.RawMessage, res *ResourcesContainer) error
...
+	// parse the "id" section
+	id, ok := m["id"]
+	if !ok {
+		return nil, errors.New(iksParsingIssue)
+	}
+	idStr, err := json.Marshal(id)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(idStr, &res.ID)
```
[the remark this answers is not in the document]
Comment (adisos) [reply pkg/ibmvpc/parser.go]: done

File: pkg/ibmvpc/parser.go [hunk 5]
```diff
@@ -443,6 +531,35 @@ func getNACLconfig(rc *ResourcesContainer, res *vpcmodel.CloudConfig, subnetName
...
+		if subnet.cidr == cidr {
+			return subnet, nil
+		}
+	}
+	return nil, fmt.Errorf("could not find subnet with cidr: %s", cidr)
+}
+
+func getIKSnodesConfig(res *vpcmodel.CloudConfig, subnetNameToSubnet map[string]*Subnet, rc *ResourcesContainer) error {
+	for i := range rc.iksNodes {
+		iksNode := rc.iksNodes[i]
```
[the remark this answers is not in the document]
Comment (adisos) [reply pkg/ibmvpc/parser.go]: done

**Review by zivnevo — APPROVED** [review]
LGTM

**Pull request merged**